client/my-sites/email/email-management/titan-management-nav/index.jsx

```diff
@@ -152,6 +167,15 @@ class TitanManagementNav extends React.Component {
 		);
 	};
 
+	recordNavItemClick = ( clickedItem ) => {
+		const { domain } = this.props;
+
+		recordTracksEvent( 'calypso_email_management_titan_nav_item_click', {
+			clicked_item: clickedItem,
+			domain: domain.name,
+		} );
+	};
```

Comment:
Maybe we could return a closure here like this:

```suggestion
	recordNavItemClick = ( clickedItem ) => {
		return () => {
			const { domain } = this.props;

			recordTracksEvent( 'calypso_email_management_titan_nav_item_click', {
				clicked_item: clickedItem,
				domain: domain.name,
			} );
		};
	};
```

Then we could simply call it like:
```
onClick={ this.recordNavItemClick( 'manage_purchase' ) }
```

Reply:
Nice! Note we could probably simplify this function even more:

```javascript
recordNavItemClick = ( clickedItem ) => () => {
  recordTracksEvent( 'calypso_email_management_titan_nav_item_click', {
    clicked_item: clickedItem,
    domain: this.props.domain.name,
  } );
};
```

Reply:
Thanks for the suggestions! I've used the multi-function approach suggested by @stephanethomas (though I kept the `const { domains }` declaration -- I just prefer that!).

The other thing I did for clarity is to rename this function to `getNavItemClickHandler()`, as I want it to be super clear that return type is a function. When I saw `this.recordNavItemClick( 'something' )` as the event handler, it just didn't feel right to me!